cts/cli/regression.tools.exp

```diff
@@ -5514,6 +5514,9 @@ Removing constraint: cli-prefer-dummy
   <change operation="delete" path="/cib/configuration/comment" position="1"/>
   <change operation="delete" path="/cib/configuration/resources/comment" position="0"/>
   <change operation="delete" path="/cib/configuration/resources/primitive[@id='Fencing']/operations/op[@id='Fencing-start-0']"/>
+  <change operation="create" path="/cib/configuration/crm_config/cluster_property_set[@id='cib-bootstrap-options']" position="0">
+    <!-- test: move attribute "value" before "name" -->
+  </change>
```

Comment:
This means we missed something. This comment is unchanged, so it shouldn't show up in a diff. There must be some spot where we still need to process non-elements.

Reply:
FYI the command being run here is `crm_diff -o $test_home/cli/crm_diff_old.xml -n $test_home/cli/crm_diff_new.xml`

Reply:
It's this hunk in `lib/common/xml.c`:

```
--- a/lib/common/xml.c
+++ b/lib/common/xml.c
@@ -325,8 +325,8 @@ accept_attr_deletions(xmlNode *xml)
     pcmk__xe_remove_matching_attrs(xml, pcmk__marked_as_deleted, NULL);
 
     // Recursively do the same for this XML node's children
-    for (xmlNodePtr cIter = pcmk__xml_first_child(xml); cIter != NULL;
-         cIter = pcmk__xml_next(cIter)) {
+    for (xmlNodePtr cIter = pcmk__xe_first_child(xml); cIter != NULL;
+         cIter = pcmk__xe_next(cIter)) {
         accept_attr_deletions(cIter);
     }
 }
```

I thought we were safe there due to the previous line being `pcmk__xe_remove_matching_attrs`, but clearly that is not the case.  That doesn't make me feel great about the rest of the patches (though, regression and ctslab tests are fine).

I've dropped this hunk.

Reply:
Hmm, I don't see how that could matter. :( This will need some closer tracing ...

Reply:
`accept_attr_deletions()` does reset the private flags to 0, maybe that is relevant. I suspect that's actually not directly related to accepting attribute deletions but is just something else that needs to be done for `xml_accept_changes()`